src/mrpro/algorithms/reconstruction/Reconstruction.py

```diff
@@ -18,3 +38,68 @@ def forward(self, kdata: KData) -> IData:
     def __call__(self, kdata: KData) -> IData:
         """Apply the reconstruction."""
         return super().__call__(kdata)
+
+    def recalculate_fourierop(self, kdata: KData) -> Self:
+        """Update (in place) the Fourier Operator, e.g. for a new trajectory.
+
+        Also recalculates the DCF.
+
+        Parameters
+        ----------
+        kdata
+            KData to determine trajectory and recon/encoding matrix from.
+        """
+        self.fourier_op = FourierOp.from_kdata(kdata)
+        self.dcf = DcfData.from_traj_voronoi(kdata.traj)
+        return self
+
+    def recalculate_csm_walsh(self, kdata: KData, noise: KNoise | None | Literal[False] = None) -> Self:
+        """Update (in place) the CSM from KData using Walsh.
+
+        Parameters
+        ----------
+        kdata
+            KData used for pseudo-inverse reconstruction, which is then used for
```

Comment:
pseudo inverse has a very specific meaning. For multi-coil acquisitions with enough sampling point, I think "pseudo-inverse" is correct, while for truly undersampled acquisitions, I doubt that we mean the pseudo-inverse but probably just the adjoint?